### DISCUSSION TITLE
XDrip BG raising/falling fast time range?

### DISCUSSION THREAD

**Discussion by random-guy-12345** [NightscoutFoundation/xDrip#3389] [Q&A] [answered]
- What's the time range for BG falling/rising fast alerts? Is it every 5 minutes? Or rather 10/15 minutes? 
- Once alert triggers, how much time must pass so it shows up again? 
- Is it configurable somewhere?

I have the following problem. Very often my "rising/falling fast" alerts just won't work, even though XDrip shows like -10 delta for the last 5 minutes. Sometimes they are raised, but even if I keep dropping, they don't raise anymore. Hence the questions above. I need more details on how the alerts work and whether the stuff above is configurable.

Thanks

**Comment by Navid200** [comment]
Where you go to enable this alert, you are also given the option to adjust at what rate it will trigger.
Please have a look at that and if it does not answer your questions, please let me know.

**Reply by random-guy-12345** [reply]
Unfortunately it doesn't. I'm aware of the rate.
The problem is, despite of the rate, I sometimes get the alert while sometimes I don't and I was wondering why would that be. I'm trying to find a pattern.
For example, currently I have the "BG falling fast" alert set at 1 mg/dL/min. I've been dropping consistently for almost 2 hours now, so I've got 24 dots on the graph, hence 24 5-minute deltas. Out of these 24, at least 10 of them were higher (actually lower, since it's a drop) than 1/mg/dL/min (so let's say -6.4, -7.2, -8.1, -7.4, -6.5, etc.), but I've only got alerted twice. I would expect to get as many alerts as there were over-the-threshold drops but that doesn't seem to be the case.

Therefore I started wondering if:
- the alert is computed for 5-minute deltas, or maybe on a longer time period, like 3x5-minute deltas or something
- what needs to be happen so I get re-alerted. Does my glucose has to stop falling and then start falling again for the alert to trigger?

My main use case here is that if I'm dropping fast, but that's only on one delta, I can kind of ignore that. But if I keep dropping (so more over-the-threshold deltas), I'd ideally like to get realerted to know that the trend persists.
I can't see that being configurable anywhere (neither the time period of the drop on which the threshold is computed, nor the option to re-alert me).

**Reply by Navid200** [reply]
The rate should last for more than one pair of readings for it to trigger.

Your questions suggest that you may not be aware of the purpose of this alert and the alternatives.

Are you aware that we also have a forecast low and a persistent high alert?
If yes, can you explain exactly what you hope to accomplish?

**Reply by Navid200** [reply]
I'm going to add some more here hazarding the chance of being told "I knew that".

I am a user myself.  The value alone may not be enough for being protected against lows (or even highs).
That's why we have a forecast low.  It takes both the level and the trend into account.
In other words, if I am high, I want my glucose to drop fast.  In that case, a sudden drop is not a bad thing and I don't need to be alerted about it.

We don't have a forecast high the moment.  And we may never have it.
In absence of that, I would like to know if I have forgotten to take insulin.  So, I have enabled rising fast alert.  It triggers if I am going up too fast.
If I want to go up too fast, I just snooze and ignore it.

**Reply by random-guy-12345** [reply]
I do realize you have the forecasted low and persistent high alerts. I probably know all the XDrip features, went throught the entire menu many, many times :).

> The rate should last for more than one pair of readings for it to trigger.

Does it mean it has to be something like (-8, -9) for it to trigger (on 1mg/dL/minute)?

What is the rule for getting re-alerted? Does the pattern have to be broken and then appear again?
So something like (-8, -9, -2, -9, -8) to get two alerts (on 1mg/dL/minute)?

I might indeed be unaware of what was the use case you had in mind when this alert was created, I acknowledge that.

While the forecasted low is some kind of alternative to what I'm trying to achieve, it is still, well, a forecast. Not sure if it currently takes both trend and IOB and carbs eaten and all that into account, but if it does, it might or might not get triggered if wrong values are entered (e.g. carbs miscalculation) because the trend alone might be not enough to compensate. There's a lot of other stuff that can throw off the forecast too. As far as I'm aware the carb absorption rate is a fixed number and it's obviously not fixed, because it depends on IG of the food and fat/protein content of the meal. The insulin absorption also differs greatly in people (or even within the same people when injected in different sites).

I'm just looking for something more "manual". My use case is as follows: I have a problem of my sugar dropping too fast, despite of the level I'm currently at (and despite the proper insulin/carbs dosage, long story in general). Firstly, I'd like to monitor that. Secondly, I am able to catch a potential hypo much faster than the actual forecast would. If what I described was available I could just monitor the alerts instead of stare on XDrip for another 10-15-20 minutes.

Being able to set "number of deltas to trigger the alert" and something like "trigger every time" checkbox would do the job and make my life much easier. I understand it's currently unavailable, just wanted to make sure I didn't miss anything.

**Reply by Navid200** [reply, marked as answer]
This has come up before.  It is unfortunate.  
I'm not sure how we can help users realize the the word forecast does not mean it is a simulation.  It is not.
It only looks at the trend and the current value and extends the line to see when it will reach the trigger.  If the past few readings form a curve instead of a line, it interpolates.  So, it does not just extend a line, but rather extends the curve.

It takes the trend and the current value into account.
So, if you are already very close to low, it tells you that you will be low sooner than if you are high with the same trend.

If you are high, it will tell you you will be low sooner of the trend is very steep compared to if the trend was shallow.

The rising fast or falling fast alert only looks at the trend.  If your glucose is changing faster than the trigger, it alerts you regardless of the value.  So, if you are very high and you are dropping fast, it will alert you even though you should want to drop as fast as possible if you are really high.

You have asked a question several times that I have not answered.  Will it trigger again.  The fact that you are asking that question worries me that you are looking for the wrong thing.
The alert will trigger when it meets its conditions.
However, if you are asking when will it alert again after you snooze it, I am not sure.  I have never wondered about that.

It is a shame that many misunderstand the forecast low alert and think that it has something to do with carbs or insulin.  It does not.  It has nothing to do with any of that.
I admit that the word forecast can be confusing.  

My understanding is that even the Dexcom app may have a forecast low.  And it has nothing to do with carbs or insulin.

I would like to know how we can improve the menus to help users better appreciate the forecast low alert.

**Reply by Navid200** [reply]
So, you may say what is then the point of fast rising or fast falling alert.
This is only my use case.
My glucose should never change too fast under normal conditions.  I inject insulin before eating to avoid that.  I take a snack before starting an exercise to avoid that.
But, we are not perfect and we make mistakes.  When I make a mistake and glucose changes too fast, I need to know so that I will take action.
It has happened that I have taken s bolus and eaten, but the bolus I took was not enough for what I ate.  I get a rising fast alert.  That tells me to take a little bit more insulin.

Nothing I say here should be taken as medical advice.  You should not try to imitate what I have described here.  You need to always consult with your doctor about your practice of how you control your diabetes.
Everything I say here is meant to help me explain how xDrip can be used.  Not how diabetes should be controlled.

**Reply by random-guy-12345** [reply]
I see. I'll try the forecasted low alarm then. Thanks for explaining.

I believe the misunderstanding comes from two things:
* You have the predictive simulation feature, so people probably think it's related to that (that was actually my case).
* (related to the above actually) the "forecast" is not the best word to use IMO. In most sources you find, e.g. [here](https://plat.ai/blog/difference-between-prediction-and-forecast/) but actually all over the internet as well, a forecast is actually a prediction, just a more specific one (time series related). I come from machine learning background and the word "forecast" is used almost interchangeably with "prediction", and rightfully so. And since forecast and prediction are closely related, people connote it with the predictive simulation feature. 

Kinda hard to come up with a good word though, since forecast is actually a good word (it is a forecast after all, just a simple one), it just doesn't make it obvious that it's a trend-only feature. "Extrapolated low" alert sounds a bit too crazy, but at least doesn't have the "prediction" or "forecast" part. "Incoming low alert" - that I would automatically connote with predictive simulation as well. Perhaps something more elaborate is required, like "Future low alert (delta based)" or something.

**Reply by random-guy-12345** [reply]
No worries, I know it's not a medical advice forum. The purpose you described fits perfectly to the potential extensions that I mentioned though. We're not perfect, and CGMs are not perfect as well, so a single fast drop/rise alert can almost always be safely ignored. Some people might be fine with ignoring one, some with three. And some people might benefit from getting re-alerted faster if it turns out the trend remains (and is not only a CGM glitch for example). You might have bolused just right but had some simple sugars in your meal, so you've got a one big bump of sugar increase but that's it. But if you have underbolused though, you'd get more bumps. You might want to know, even if you're not high just yet (and not persistently high).

I didn't come here to ask for a new feature though, so I'm not going to argue about it anymore. Thanks for all the help and explanations

**Reply by Navid200** [reply]
Thanks so much for the suggestions.
I have just discovered that Dexcom calls it something else:
Urgent low soon alert

https://www.dexcom.com/en-us/faqs/how-does-the-urgent-low-soon-alert-work

Let me see what the developers feel about renaming the alert.

**Reply by random-guy-12345** [reply]
Low soon sounds good, at least it doesn't bring the predictions to mind that much.

**Reply by Navid200** [reply]
I looked at the code.

The rising or falling fast alert only triggers if two conditions are met.
First, the slope of the current step must be greater than half of the set threshold.
Second, the overall slope of the past 3 consecutive steps must be greater than the threshold.

This means that a consistent fast change will trigger the alert.  But, a fast change that is gradually flattening will not.

The threshold can be set to one of 3 different values.
Other than that, nothing else can be changed by the user.
